### PR TITLE
[NOMERGE] New package: purescript-0.14.0

### PR DESCRIPTION
**Pull request by waynevanson**
#### Summary
- Adds the wonderful compiler for purescript.
- Adds a dependency for spago (https://github.com/void-linux/void-packages/issues/24682)

___Not to be merged until___:
- [ ] GHC has been updated to 8.10.4 (https://github.com/void-linux/void-packages/pull/29506)
- [ ] Commits from adding future GHC version have been removed from this git branch
- [ ] Retested against an up-to-date master containing GHC-8.10.4

#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [x] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/purescript/template [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'purescript'
+pkgname=purescript
+version=0.14.0
+revision=1
+wrksrc="purescript-${version}"
+build_style=haskell-stack
+makedepends="ncurses-libtinfo-devel ncurses-devel zlib-devel"
+short_desc="Strongly typed language that compiles to javascript"
+maintainer="Wayne Van Son <waynevanson@gmail.com>"
+license="GPL-3.0-or-later"
```
Comment (BlueChrystal) [inline srcpkgs/purescript/template:10]: I took a look, and the PureScript compiler is licensed under the 3-clause BSD license, not GPLv3 or later.

File: srcpkgs/purescript/template [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'purescript'
+pkgname=purescript
+version=0.14.0
+revision=1
+wrksrc="purescript-${version}"
+build_style=haskell-stack
+makedepends="ncurses-libtinfo-devel ncurses-devel zlib-devel"
```
Comment (BlueChrystal) [inline srcpkgs/purescript/template:7]: Apparently, the libtinfo dependency has been removed (see https://github.com/purescript/purescript/pull/4069). The change still hasn't been released though.

File: srcpkgs/purescript/template [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'purescript'
+pkgname=purescript
+version=0.14.0
```
Comment (BlueChrystal) [inline srcpkgs/purescript/template:3]: Maybe update this to 0.14.1?

**Review by BlueChrystal — CHANGES_REQUESTED** [review]

**Comment by waynevanson** [comment]
I'm no longer working on this, so closing. Others are free to reference this :)

**Pull request closed without merging**